File: objective-c/vcdiff/Project/block.m


#import "block.h"

//definition of block class
@implementation block
-(void) initwith:(NSMutableString *) str:(int) off //allocates and initialises the members with the given values
{
	text=[NSMutableString alloc];
	text=str;
	offset=off;
	
}
-(NSMutableString *) gettext    //Returns text
{
	return text;
}

-(int) getoffset
{
	return offset;
}

@end

File: java-old/com.neoba.dsync.vcdiff/src/com/neoba/dsync/vcdiff/Block.java
package com.neoba.dsync.vcdiff;

public class Block {

    String text;
    int offset;
    
    Block(String text, int offset) {
        this.text = text;
        this.offset = offset;
    }

    String getText() {
        return this.text;
    }

    int getOffset() {
        return this.offset;
    }
}
